mongosql/src/ast/rewrites/aggregate.rs

```diff
@@ -165,6 +174,59 @@ impl Visitor for AggregateUsageCheckVisitor {
     }
 }
 
+#[derive(Default)]
+pub struct MultiArgCountVisitor {
+    pub error: Option<Error>,
+}
+
+impl Visitor for MultiArgCountVisitor {
+    fn visit_group_by_clause(&mut self, node: GroupByClause) -> GroupByClause {
+        let aggregations = node
+            .aggregations
+            .into_iter()
+            .map(|agg_func| match agg_func.expr {
+                ast::Expression::Function(ast::FunctionExpr {
+                    function: ast::FunctionName::Count,
+                    args: ast::FunctionArguments::Args(args),
+                    set_quantifier,
+                }) if args.len() > 1 => {
+                    let doc: Vec<ast::DocumentPair> = args
+                        .into_iter()
+                        .map(|arg| {
+                            let ast::Expression::Identifier(key) = arg.clone() else {
+                                self.error = Some(Error::InvalidMultiArgCountArg);
```

Comment:
I decided for now we will only support field references as the multiple arguments. It is doubtful anyone will want to do something like `COUNT(a+1, WHEN b CASE 7 THEN "yes" ELSE c END, x)`. If someone requests this, we can support it then. For now, this is a very convenient restriction.

Reply:
Yeah that's a good call, no need to add too much complexity if it's not requested.